lib/ansible/runner/credential/machine_credential.rb

```diff
@@ -6,13 +6,15 @@ def self.auth_type
       end
 
       def command_line
-        {:user => auth.userid}.delete_blanks.merge(become_args)
+        {:user => auth.userid}.delete_blanks.merge(become_args).tap do |args|
+          args[:ask_pass] = nil if auth.password.present?
```

Comment:
Gave a suggestion for using `.delete` instead so the previous specs don't fail:

https://github.com/carbonin/manageiq/pull/5/commits/9ff7d6593cf3120ef398b9aa83846a6dae6e0f14

We aren't mutating because of the `.merge` call that precedes the `.tap`, so this should be fine.

Reply:
I don't actually want to use `delete` here. AwesomeSpawn interprets a hash key with an explicit nil value as `--ask_pass` without a passed value (so a flag command line argument).

Specifically this behavior https://github.com/ManageIQ/awesome_spawn/blob/fecd068cc2fad16f85d9c80c59767b84b468474f/lib/awesome_spawn/command_line_builder.rb#L36

Reply:
Oh, okay.  Then yeah, the specs just need to be updated to reflect that new addition (about 4 of them were failing because this key now exists).

Reply:
Done